src/test/java/com/amazonaws/services/kinesis/clientlibrary/config/KinesisClientLibConfiguratorTest.java

```diff
@@ -329,6 +329,29 @@ public void testWithAWSCredentialsFailed() {
         }
     }
 
+    @Test
+    public void testWithDifferentAWSCredentialsForDynamoDBAndCloudWatch() {
```

Comment:
This test appears to succeed with the current code.  It might make sense to test with different class names for each credential provider.  The credential decoder has the ability to load [custom class names](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/clientlibrary/config/AWSCredentialsProviderPropertyValueDecoder.java#L97).